lib/features/docker_save.js

```diff
@@ -23,26 +27,81 @@ export default class DockerSave {
     let imgStream = await image.get();
     let zipStream = zlib.createGzip();
     imgStream.pipe(zipStream).pipe(fs.createWriteStream(pathname));
-    await waitForEvent(zipStream, 'end');
+    await new Promise((accept, reject) => {
+      zipStream.on('end', accept);
+      zipStream.on('error', (error) => reject(error));
+    });
     debug('tar written');
 
     let stat = await fs.stat(pathname);
     debug(stat.size);
     let uploadStream = fs.createReadStream(pathname);
-    let expiration = new Date(Date.now() + task.runtime.dockerSave.expiration);
+    let expiration = taskcluster.fromNow(task.runtime.dockerSave.expiration);
 
-    await uploadToS3(task, uploadStream, ARTIFACT_NAME, expiration, {
+    await uploadToS3(task, uploadStream, 'public/dockerImage.tar', expiration, {
       'content-type': 'application/x-tar',
       'content-length': stat.size,
       'content-encoding': 'gzip'
     });
 
-    debug('artifact uploaded');
+    debug('docker image uploaded');
 
     //cleanup
-    fs.unlink(pathname).catch(() => {
-      task.runtime.log('[alert-operator] could not delete docker save tarball, worker may run out of hdd space');
+    fs.unlink(pathname).catch((err) => {
+      task.runtime.log('[alert-operator] could not delete cache save tarball, worker may run out of hdd space\r\n'
+        + err + err.stack);
     });
     await image.remove();
   }
+
+  //uploading a cache folder as an artifact
+  async uploadCache(task, cacheStr) {
+    let cache = cacheStr.split(KEY_DELIMITER);
+    let loc = path.join(task.runtime.cache.volumeCachePath, cache[0], cache[1] + '/');
+
+    //temporary path for saved file
+    let pathname = path.join('/tmp', slugid.v4() + '.tar');
```

Comment:
We might want to look at storing this somewhere under /mnt.  The AWS instances are created with 8gb root volumes, and then 80+GB mounted under /mnt.

Here is a sample from one of our instances:
$ df -H
/dev/xvda1                        8.4G  1.8G  6.1G  23% /
none                              4.1k     0  4.1k   0% /sys/fs/cgroup
udev                              7.9G   13k  7.9G   1% /dev
tmpfs                             1.6G  1.2M  1.6G   1% /run
none                              5.3M     0  5.3M   0% /run/lock
none                              7.9G  1.3M  7.9G   1% /run/shm
none                              105M     0  105M   0% /run/user
/dev/mapper/instance_storage-all   80G   22G   54G  29% /mnt